### ISSUE TITLE
Nullable list generated as an empty list

### ISSUE THREAD

**Issue by TanguyChevillardNovade** [gql-dart/ferry#219]
Hello,
One of my GraphQL types has two fields that can either be null or non-empty lists:
![image](https://user-images.githubusercontent.com/61184332/124921430-aea71100-dff8-11eb-8c03-f0c1eab09d83.png)

My issue is the fact that, in the type generated by Ferry, those fields are not nullable but instead created as empty lists:

![image](https://user-images.githubusercontent.com/61184332/124921752-047bb900-dff9-11eb-8e0e-ab318b7721fe.png)

This causes errors in my backend, as empty lists are not expected there.

I have created an example in this repo:
https://github.com/TanguyChevillardNovade/FerryEmptyList

Is this a bug, or did I forget to add something in one of the GraphQL files?

**Comment by smkhalsa** [comment]
This is likely caused by https://github.com/google/built_value.dart/issues/1049 which I believe was resolved in [v8.1.0](https://github.com/google/built_value.dart/blob/master/CHANGELOG.md#810).

We may need to use the new `serializeNulls` option, but I haven't looked at it very closely.

PRs welcome.

**Comment by GP4cK** [comment]
@smkhalsa I was looking into this issue to potentially make a PR. If we use serializeNulls without nothing else, I'm afraid the JSON output will have all the nullable fields without value set to null. However, I would like to only set the ones I chose to null (the others should not appear in the JSON).

I know that drift has a class [Value](https://github.com/simolus3/drift/blob/49668e3a578934f5e717bd93db5a0714634edec5/drift/lib/src/runtime/data_class.dart#L128) which they use to differentiate between:
1. a value being absent 
2. the value being null.

Would it be ok with you if I do the same? However, this may introduce a breaking change because this code:

```dart
final reviewsReq = GReviewsReq((b) => b.vars.first = 10);
```
will become:
```dart
final reviewsReq = GReviewsReq((b) => b.vars.first = Value(10));
```
if `first` is nullable.

Is that ok with you? Or do you see an alternative?

**Comment by smkhalsa** [comment]
I'd rather avoid this.

Is the issue of absent vs nullable only present when converting to JSON?

**Comment by GP4cK** [comment]
> Is the issue of absent vs nullable only present when converting to JSON?

I think so. At least for my use case and I believe the one of the issue.

**Comment by smkhalsa** [comment]
The original issue relates to the Dart classes allowing lists with null values.

**Comment by GP4cK** [comment]
Mmm I thought the original issue was to be able to not set `in` and `notIn` (or set it to absent) so that the client does not send an empty list to the backend.
Anyways, do you have a better idea to distinguish between absent vs null? I'd be happy to help if needed.

**Comment by smkhalsa** [comment]
Does built_value's own internal representation of the data distinguish between null and absent?

**Comment by GP4cK** [comment]
I don't think so. I've made a sample repo: https://github.com/GP4cK/built_value_absent
You can run it with `dart run` and it will output in the console the JSON with null (even though I did not specify a value).

**Comment by GP4cK** [comment]
Basically whether you set `serializeNulls` to true or false, it will either include null in the JSON or ignore it but it doesn't make the difference between null and absent.

**Comment by GP4cK** [comment]
I also created an issue on built_value's repo: https://github.com/google/built_value.dart/issues/1123
